backend/src/main/scala/cromwell/backend/WorkflowBackendActor.scala

```diff
@@ -29,37 +31,68 @@ object WorkflowBackendActor {
   case class AbortSucceeded(jobKey: BackendJobDescriptorKey) extends AbortResponse
   case class AbortFailed(jobKey: BackendJobDescriptorKey, throwable: Throwable) extends AbortResponse
+
+  sealed trait InitializationResponse extends WorkflowBackendActorResponse
+  case object InitializationSuccess extends InitializationResponse
+  case class InitializationFailed(reason: Throwable) extends InitializationResponse
 }
 
 /**
   * Workflow-level actor for executing, recovering and aborting jobs.
   */
 trait WorkflowBackendActor extends Actor with ActorLogging {
 
+  /**
+    * The execution context for the actor
+    */
   protected implicit def ec: ExecutionContext
+
+  /**
+    * The set of calls which this backend will be expected to run
+    */
+  protected def calls: Seq[Call]
+
+  /**
+    * The workflow descriptor for the workflow in which this Backend is being used
+    */
   protected def workflowDescriptor: BackendWorkflowDescriptor
+
+  /**
+    * The configuration for the backend, in the context of the entire Cromwell configuration file.
+    */
   protected def configurationDescriptor: BackendConfigurationDescriptor
 
   def receive: Receive = LoggingReceive {
-    case Execute(jobDescriptor) => performActionThenRespond(execute, jobDescriptor, onFailure = descriptorToExecutionFailure)
-    case Recover(jobDescriptor) => performActionThenRespond(recover, jobDescriptor, onFailure = descriptorToExecutionFailure)
-    case Abort(jobKey)          => performActionThenRespond(abort, jobKey, onFailure = AbortFailed)
+    case Initialize             => performActionThenRespond(initSequence, onFailure = InitializationFailed)
+    case Execute(jobDescriptor) => performActionThenRespond( () => execute(jobDescriptor), onFailure = executionFailed(jobDescriptor.key))
+    case Recover(jobDescriptor) => performActionThenRespond( () => recover(jobDescriptor), onFailure = executionFailed(jobDescriptor.key))
+    case Abort(jobKey)          => performActionThenRespond( () => abort(jobKey), onFailure = abortFailed(jobKey))
+  }
+
+  override def postStop = {
+    afterAll()
   }
 
-  private def performActionThenRespond[A](operation: A => Future[WorkflowBackendActorResponse],
-                                          operationArgument: A,
-                                          onFailure: (A, Throwable) => WorkflowBackendActorResponse) = {
-    // To avoid sending a response to the wrong `sender` if more than one `execute` arrives before the first one finishes,
-    // for example:
-    val sndr = sender()
-    operation(operationArgument) onComplete {
-      case Success(r) => sndr ! r
-      case Failure(t) => sndr ! onFailure(operationArgument, t)
+  private def performActionThenRespond(operation: () => Future[WorkflowBackendActorResponse],
+                                       onFailure: (Throwable) => WorkflowBackendActorResponse) = {
+    val respondTo: ActorRef = sender
+    operation.apply onComplete {
```

Comment:
The Akka devs frequently advise against using callbacks on Futures inside of an Actor like this. At the moment all of the alternate schemes I've thought of would incur more of a hassle on the part of a backend implementer (although one could argue that requiring the implementer to be sure to not F themselves by screwing up their internal state is just as bad). 

I'm still looking at alternatives but if someone else sees this and has a good idea please chime in.